lib/dnf.py

```diff
@@ -12,41 +12,29 @@
 _Repo = collections.namedtuple('Repo', ['name', 'baseurl', 'data', 'file'])
 
 
-# dnf up to dnf4
 def _get_repos_dnf():
-    db = dnf.Base()
-    # black magic to make the dnf python API load /etc/dnf/dnf.conf
-    # and variables from /etc/dnf/vars/*, used by CentOS Stream to
-    # define $stream, used in metalinks
-    # source: https://bugzilla.redhat.com/show_bug.cgi?id=1920735#c2
-    db.conf.read(priority=dnf.conf.PRIO_MAINCONFIG)
-    db.conf.substitutions.update_from_etc(installroot=db.conf.installroot, varsdir=db.conf.varsdir)
-    db.read_all_repos()
-    for name, repo in db.repos.items():
-        # no disabled repos
-        if not repo.enabled:
-            continue
-        repo.load()
-        baseurl = repo.remote_location('/')
+    cmd = util.libdir / 'dnf_get_repos'
+    ret = util.subprocess_run(cmd, check=True, stdout=subprocess.PIPE, universal_newlines=True)
+    try:
+        json_data = json.loads(ret.stdout)
+    except json.JSONDecodeError as e:
+        util.log(f"Failed to decode JSON data from {cmd}: {e}")
+        return
+
+    for i in json_data:
```

Comment:
Let's not use single-letter variables on this big scope.

Maybe?
```suggestion
    for repo in json_data:
```